### PR TITLE
bgpv1,ci: Add Test_AdvertisedPathAttributes into BGP component tests

### PR DESCRIPTION
**Pull request by rastislavs**
Adds a new BGP component test which tests the `AdvertisedPathAttributes` options of the `CiliumBGPPeeringPolicy` CRD. As iBGP peering is needed to test Local Preference, it modifies the test framework to allow it.

### CODE REVIEW

File: pkg/bgpv1/test/adverts_test.go [hunk 1]
```diff
@@ -622,3 +627,188 @@ func Test_LBEgressAdvertisement(t *testing.T) {
...
+					}),
+				},
+			},
+		},
+	}
+
+	testCtx, testDone := context.WithTimeout(context.Background(), maxTestDuration)
+	defer testDone()
+
+	// setup topology - iBGP (ASN == ciliumASN)
```
Comment (harsimran-pabla) [inline pkg/bgpv1/test/adverts_test.go:712]: Why use iBGP for this test? Are some of these communities non-transitive between AS?

Comment (rastislavs) [reply pkg/bgpv1/test/adverts_test.go:712]: To test the Local Preference Attribute. That is not advertised to eBGP peers.

**Review by harsimran-pabla — APPROVED** [review]

**Review by ldelossa — APPROVED** [review]

**Pull request merged**